src/main/groovy/com/morpheusdata/nutanix/prismelement/plugin/NutanixPrismElementImageStoreDatasetProvider.groovy

```diff
@@ -0,0 +1,116 @@
+package com.morpheusdata.nutanix.prismelement.plugin
+
+import com.morpheusdata.core.MorpheusContext
+import com.morpheusdata.core.Plugin
+import com.morpheusdata.core.data.DataQuery
+import com.morpheusdata.core.data.DatasetInfo
+import com.morpheusdata.core.data.DatasetQuery
+import com.morpheusdata.core.providers.AbstractDatasetProvider
+import com.morpheusdata.model.Datastore
+import io.reactivex.rxjava3.core.Observable
+
+import javax.sql.DataSource
+
+class NutanixPrismElementImageStoreDatasetProvider extends AbstractDatasetProvider<Datastore, String> {
+    public static final PROVIDER_NAME = "Nutanix Image Store Provider"
+    public static final PROVIDER_NAMESPACE = "com.morpheusdata.nutanix.prismelement.plugin"
+    public static final PROVIDER_KEY = "nutanixContainers"
+    public static final PROVIDER_DESCRIPTION = "The default image store to use with Nutanix Prism Element"
+
+    NutanixPrismElementImageStoreDatasetProvider(Plugin plugin, MorpheusContext context) {
+        this.plugin = plugin
+        this.morpheusContext = context
+    }
+
+    /**
+     * {inheritDoc}
+     */
+    @Override
+    DatasetInfo getInfo() {
+        new DatasetInfo(
+                name: PROVIDER_NAME,
+                namespace: PROVIDER_NAMESPACE,
+                key: PROVIDER_KEY,
+                description: PROVIDER_DESCRIPTION,
+        )
+    }
+
+    /**
+     * {inheritDoc}
+     */
+    @Override
+    Class<Datastore> getItemType() {
+        return Datastore.class
+    }
+
+    /**
+     * {inheritDoc}
+     */
+    @Override
+    Observable<Datastore> list(DatasetQuery query) {
+        morpheusContext.async.cloud.datastore.list(
+                new DataQuery()
+                        .withFilter('refType', 'ComputeZone')
+//                        .withFilter('refId', zoneId) // TODO: how do I do this, no params to pull from?
```

Comment:
The original was an option source that pulled this from `params`. I'm not sure how to replicate this in a DatasetProvider. Any ideas?